core/ts.cpp

```diff
@@ -385,6 +386,21 @@ void TransitionSystem::add_inputvar(const Term & v)
   name_term(v->to_string(), v);
 }
 
+Term TransitionSystem::promote_inputvar(Term iv)
+{
+  assert(iv);
+  size_t num_erased = inputvars_.erase(iv);
+  if (!num_erased) {
+    throw PonoException("Cannot promote non-inputvars: " + iv->to_string());
+  }
+
+  assert(iv);
```

Comment:
we have a similar assert above. Is this needed?

Reply:
Good point addressed with be917378730b7130df2ccc9959c8037134a79fe7